### DISCUSSION TITLE
In Flutter firebase_ui_auth package how to access to register or login screen's textfields inputs in SignInScreen when using Email/password provider

### DISCUSSION THREAD

**Discussion by husain-nahar** [firebase/flutterfire#10015] [Q&A]
I am using flutter firebase_ui_auth package, and I have successfully implemented the SIgnInScreen flow however I have a very simple question that how can I get access/control over user inputs inside textfields of register/login view of Email/password auth provider of firebase. As I want to get those values like email password so that I can send it to my backend server as well.

I tried to investigate the package but nothing so far worked.

**Comment by brendanatlifewise** [comment]
I have this same issue for password.  
@husain-nahar you can get the email from the current user.  `FirebaseAuth.instance.currentUser.email`

You can also find it in the callback arguments for the actions
`SignInScreen(
actions: [
AuthStateChangeAction<SignedIn>((context, state) async {
          var email = state.user.email;
 }),`

I need a way to access the password, however.  And naturally the TextEditingController is buried in the package and is a private field.  This could be a reason to abandon the package unless there is a resolution.

**Reply by husain-nahar** [reply]
Thanks for the suggestion. 
however I am already getting the email input by two different ways same as you suggested above. 
But stuck with getting the password field input.
Someone from the Flutter should reply to this I request.
For now I am going all manually for the email/password as well as social auth login sign up flow.
